homeassistant/const.py

```diff
@@ -149,6 +149,7 @@ class Platform(StrEnum):
 CONF_EVENT_DATA_TEMPLATE: Final = "event_data_template"
 CONF_EXCLUDE: Final = "exclude"
 CONF_EXTERNAL_URL: Final = "external_url"
+CONF_EXTRA: Final = "extra"
```

Comment:
Let's call it `metadata`. Also, no need to put in main const file, as we should just allow validation it, but we should never try to read it out